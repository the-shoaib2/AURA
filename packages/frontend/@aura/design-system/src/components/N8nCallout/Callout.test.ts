import { render } from '@testing-library/vue';

import auraCallout from './Callout.vue';

describe('components', () => {
	describe('auraCallout', () => {
		it('should render info theme correctly', () => {
			const wrapper = render(auraCallout, {
				props: {
					theme: 'info',
				},
				global: {
					stubs: ['aura-icon', 'aura-text'],
				},
				slots: {
					default: '<aura-text size="small">This is an info callout.</aura-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render success theme correctly', () => {
			const wrapper = render(auraCallout, {
				props: {
					theme: 'success',
				},
				global: {
					stubs: ['aura-icon', 'aura-text'],
				},
				slots: {
					default: '<aura-text size="small">This is a success callout.</aura-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render warning theme correctly', () => {
			const wrapper = render(auraCallout, {
				props: {
					theme: 'warning',
				},
				global: {
					stubs: ['aura-icon', 'aura-text'],
				},
				slots: {
					default: '<aura-text size="small">This is a warning callout.</aura-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render danger theme correctly', () => {
			const wrapper = render(auraCallout, {
				props: {
					theme: 'danger',
				},
				global: {
					stubs: ['aura-icon', 'aura-text'],
				},
				slots: {
					default: '<aura-text size="small">This is a danger callout.</aura-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render secondary theme correctly', () => {
			const wrapper = render(auraCallout, {
				props: {
					theme: 'secondary',
				},
				global: {
					stubs: ['aura-icon', 'aura-text'],
				},
				slots: {
					default: '<aura-text size="small">This is a secondary callout.</aura-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render custom theme correctly', () => {
			const wrapper = render(auraCallout, {
				props: {
					theme: 'custom',
					icon: 'git-branch',
				},
				global: {
					stubs: ['aura-icon', 'aura-text'],
				},
				slots: {
					default: '<aura-text size="small">This is a secondary callout.</aura-text>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
		it('should render additional slots correctly', () => {
			const wrapper = render(auraCallout, {
				props: {
					theme: 'custom',
					icon: 'git-branch',
				},
				global: {
					stubs: ['aura-icon', 'aura-text', 'aura-link'],
				},
				slots: {
					default: '<aura-text size="small">This is a secondary callout.</aura-text>',
					actions: '<aura-link size="small">Do something!</aura-link>',
					trailingContent:
						'<aura-link theme="secondary" size="small" :bold="true" :underline="true" to="https://aura.io">Learn more</aura-link>',
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
