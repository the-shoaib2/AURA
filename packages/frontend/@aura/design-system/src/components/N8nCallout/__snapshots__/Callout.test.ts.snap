// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > auraCallout > should render additional slots correctly 1`] = `
"<div class="aura-callout callout custom round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <aura-icon-stub icon="git-branch" size="medium" spin="false"></aura-icon-stub>
    </div>
    <aura-text-stub bold="false" size="small" compact="false" tag="span"></aura-text-stub> &nbsp; <aura-link-stub size="small"></aura-link-stub>
  </div>
  <aura-link-stub theme="secondary" size="small" bold="true" underline="true" to="https://aura.io"></aura-link-stub>
</div>"
`;

exports[`components > auraCallout > should render custom theme correctly 1`] = `
"<div class="aura-callout callout custom round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <aura-icon-stub icon="git-branch" size="medium" spin="false"></aura-icon-stub>
    </div>
    <aura-text-stub bold="false" size="small" compact="false" tag="span"></aura-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > auraCallout > should render danger theme correctly 1`] = `
"<div class="aura-callout callout danger round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <aura-icon-stub icon="triangle-alert" size="medium" spin="false"></aura-icon-stub>
    </div>
    <aura-text-stub bold="false" size="small" compact="false" tag="span"></aura-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > auraCallout > should render info theme correctly 1`] = `
"<div class="aura-callout callout info round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <aura-icon-stub icon="info" size="medium" spin="false"></aura-icon-stub>
    </div>
    <aura-text-stub bold="false" size="small" compact="false" tag="span"></aura-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > auraCallout > should render secondary theme correctly 1`] = `
"<div class="aura-callout callout secondary round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <aura-icon-stub icon="info" size="medium" spin="false"></aura-icon-stub>
    </div>
    <aura-text-stub bold="false" size="small" compact="false" tag="span"></aura-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > auraCallout > should render success theme correctly 1`] = `
"<div class="aura-callout callout success round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <aura-icon-stub icon="circle-check" size="medium" spin="false"></aura-icon-stub>
    </div>
    <aura-text-stub bold="false" size="small" compact="false" tag="span"></aura-text-stub> &nbsp;
  </div>
</div>"
`;

exports[`components > auraCallout > should render warning theme correctly 1`] = `
"<div class="aura-callout callout warning round" role="alert">
  <div class="messageSection">
    <div class="icon">
      <aura-icon-stub icon="triangle-alert" size="medium" spin="false"></aura-icon-stub>
    </div>
    <aura-text-stub bold="false" size="small" compact="false" tag="span"></aura-text-stub> &nbsp;
  </div>
</div>"
`;
